packages/common/src/api/tan-query/useCurrentUser.ts

```diff
@@ -1,48 +1,141 @@
+import { isResponseError } from '~/utils/error'
 
 import { QUERY_KEYS } from './queryKeys'
 import { Config } from './types'
 
+const fetchAccount = async (
+  args: full.GetUserAccountRequest,
+  { sdk }: { sdk: AudiusSdk }
+) => {
+  try {
+    const { data } = await sdk.full.users.getUserAccount(args)
+    if (!data) {
+      console.warn('Missing user from account response')
+      return null
+    }
+
+    const account = accountFromSDK(data)
+    // If we got a valid account, populate user bank since that's
+    // expected to exist on "account" users
+    if (account) {
+      const userBank = await sdk.services.claimableTokensClient.deriveUserBank({
+        ethWallet: args.wallet,
+        mint: 'wAUDIO'
+      })
+      account.user.userBank = userBank.toString() as SolanaWalletAddress
+    }
+    return account ?? null
+  } catch (e) {
+    // Account doesn't exist, don't bubble up an error, just return null
+    if (isResponseError(e) && [401, 404].includes(e.response.status)) {
+      return null
+    }
+    throw e
+  }
+}
+
+/**
+ * Saga-compatible version of fetching an account by wallet.
+ */
+export function* fetchAccountSaga(
+  args: full.GetUserAccountRequest,
+  options?: Config
+) {
+  const queryClient = yield* getContext('queryClient')
+  const sdk = yield* getSDK()
+
+  const result: Awaited<ReturnType<typeof fetchAccount>> = yield* call(
+    async () =>
+      queryClient.fetchQuery({
+        queryKey: [QUERY_KEYS.accountUser, args.wallet],
+        queryFn: async () => fetchAccount(args, { sdk }),
+        // Don't refetch on new mounts by default
+        staleTime: options?.staleTime ?? Infinity
+      })
+  )
+  return result
+}
+
 /**
- * Hook to get the currently logged in user's data
+ * Hook to get the full account info for the current user
  */
-export const useCurrentUser = (config?: Config) => {
+export const useAccount = (options?: Config) => {
   const { audiusSdk } = useAudiusQueryContext()
   const { currentUser } = useSelector(getWalletAddresses)
 
   return useQuery({
     queryKey: [QUERY_KEYS.accountUser, currentUser],
-    queryFn: async () => {
-      const sdk = await audiusSdk()
-      const { data } = await sdk.full.users.getUserAccount({
-        wallet: currentUser!
-      })
+    queryFn: async () =>
+      fetchAccount({ wallet: currentUser! }, { sdk: await audiusSdk() }),
+    // Don't refetch on new mounts by default
+    staleTime: options?.staleTime ?? Infinity,
+    enabled: options?.enabled !== false && !!currentUser
+  })
+}
+
+/**
+ * Hook to get the current user's metadata
+ */
+export const useCurrentUser = (options?: Config) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const { currentUser } = useSelector(getWalletAddresses)
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.accountUser, currentUser],
+    queryFn: async () =>
+      fetchAccount({ wallet: currentUser! }, { sdk: await audiusSdk() }),
+    // Don't refetch on new mounts by default
```

Comment:
Curious what we think about defaulting things to Infinity if not specified (we don't want to constantly refetch the account).
And also how we feel about this pattern of using `select` for hooks that don't need the full account data but still want to share the account in the cache.

Reply:
yeah i like infinity for the account for now. yes the select pattern is perfect for this, agreed